### ISSUE TITLE
poor PPO results on Pendulum-v0

### ISSUE THREAD

**Issue by natetsang** [PacktPublishing/Tensorflow-2-Reinforcement-Learning-Cookbook#45]
First, let me say that I've enjoyed the book and going through your implementations! I've found them intuitive and easy to follow.

I ran the `6_ppo_continuous.ipynb` for 600 epochs and it didn't appear to learn at all. The rewards fluctuated from -1500 to -1200 throughout the 600 epochs. I didn't make any changes to the code except for the epochs. Is there a mistake in the code or default hyperparameters? What reward for `Pendulum-v0` do you expect for your implementation? I've seen other working implementations with a reward of around -200 after training.

Thanks!

**Comment by praveen-palanisamy** [comment]
Hi @natetsang,

Happy to hear that you enjoyed going through this Cookbook!
PPO on continuous control problems (like in `Pendulum-v0`) does take more iterations to learn a good policy that gets good rewards. The typical number of episodes to train for `Pendulum-v0` with PPO until good performance is in the range 5000 to 10,000 depending on the hyperparameters (and the seeds), which corresponds to about 1M to 2M iterations. 
The default values for the hyperparameters in the code may not be the best but were the values I found to yield rewards in the convergence/acceptable range: >-300 IIRC.
There could be more tricks & clever hyperparameter tuning to accelerate the convergence but the implementations in this CookBook is aimed at being concise and easy-to-follow.
If you find any bugs in the code, please let me know or raise an issue. I quickly went through the code again and didn't find any obvious issues in it.

**Comment by natetsang** [comment]
Thanks @praveen-palanisamy for your response - it's very much appreciated. I didn't know it was supposed to take that long so I'll give that a try. 

I've tested multiple Pytorch implementations and they all succeed on Pendulum after training for only a couple of minutes. For instance, I ran [this](https://github.com/alejopijuan/PPO-Implementations) pytorch version and it trained very quickly on `Pendulum-v0`. 

I s it possible tensorflow isn't tracking the gradients as well. I even translated it to tensorflow line-by-line [here](https://github.com/alejopijuan/PPO-Implementations/pull/1/commits/eaabaab6d31b348d919410f7958f9a738bc51013) and it didn't train at all. It seems very odd that TF wouldn't work but Pytorch would.
